ui/imports/shared/controls/chat/ChatInputLinksPreviewArea.qml

```diff
@@ -0,0 +1,153 @@
+
+    signal imageRemoved(int index)
+    signal imageClicked(var chatImage)
+    signal linkReload(string link)
+    signal linkClicked(string link)
+    signal linkRemoved(string link)
+
+    horizontalPadding: 12
+    topPadding: 12
+
+    contentItem: Item {
+        id: opacityMaskWrapper
+        
+        anchors.fill: parent
+
+        implicitWidth: flickable.implicitWidth
+        implicitHeight: flickable.implicitHeight
+
+        opacity: 0
+
+        Flickable {
+            id: flickable
+            
+            anchors.fill: parent
+            anchors.leftMargin: root.leftPadding
+            anchors.rightMargin: root.rightPadding
+            anchors.bottomMargin: root.bottomPadding
+            anchors.topMargin: root.topPadding
+
+            implicitHeight: contentHeight
+            implicitWidth: contentWidth
+
+            contentWidth: layout.width
+            contentHeight: layout.height
+
+            RowLayout {
+                id: layout
+                spacing: 8
+                StatusChatInputImageArea {
+                    id: imageArea
+                    Layout.preferredHeight: 64
+                    spacing: layout.spacing
+                    imageSource: imagePreviewModel
+                    onImageClicked: root.imageClicked(chatImage)
+                    onImageRemoved: root.imageRemoved(index)
+                    visible: !!imagePreviewModel && imagePreviewModel.length > 0
+                }
+                Repeater {
+                    model: d.filteredModel
+                    delegate: LinkPreviewMiniCard {
+                        // Model properties
+                        required property string title
+                        required property string url
+                        required property bool unfurled
+                        required property bool immutable
+                        required property string hostname
+                        required property string description
+                        required property int linkType
+                        required property int thumbnailWidth
+                        required property int thumbnailHeight
+                        required property string thumbnailUrl
+                        required property string thumbnailDataUri
+
+                        required property int index
+
+                        Layout.preferredHeight: 64
+
+                        titleStr: title
+                        domain: hostname                //TODO: use domain when available
+                        favIconUrl: thumbnailImageUrl   //TODO: use favicon when available
+                        communityName: ""               //TODO: add community info when available
+                        channelName: ""                 //TODO: add community info when available
+
+                        thumbnailImageUrl: thumbnailDataUri.length > 0 ? thumbnailDataUri : thumbnailUrl
```

Comment:
I think it should be the opposite.

```suggestion
                        thumbnailImageUrl: thumbnailDataUri.length == 0 ? thumbnailDataUri : thumbnailUrl
```

We should always prefer `thumbnailUrl` over `thumbnailDataUri`. The URL is pointing to a local media server.
- The former will be present when the preview is part of a Message. `daraUri` will never be appear in this case.
- The latter is used when the preview is shown in MessageComposer, because there's nowhere to point the URL to. Though I think we might have a URL in this case sometimes as well.

Also, I didn't look into the message previews part, please check that it's fine there too.